Add method for getting packet flags from ENetPacketPeer

Chaosed0 opened the pull request:
Fixes proposal 10240: https://github.com/godotengine/godot-proposals/issues/10240

This PR adds the `get_packet_flags` method to `ENetPacketPeer` which can be used to obtain the flags sent with the packet. In my experimentation, it appears that not all flags are replicated from peer to peer, but it's unclear which are and which aren't (the ENet documentation does not specify), so I opted not to include this info explicitly in Godot's documentation.

This is my first PR, so let me know if I have missed anything.

*Bugsquad edit:*
- Closes https://github.com/godotengine/godot-proposals/issues/10240

AThousandShips commented on modules/enet/enet_packet_peer.cpp:
```diff
@@ -175,6 +175,10 @@ int ENetPacketPeer::get_channels() const {
 }
 
+int ENetPacketPeer::get_packet_flags() const {
+	return packet_queue.front()->get()->flags;
```
This will crash if the queue is empty, should have a check like that on line 93

AThousandShips commented on modules/enet/enet_packet_peer.cpp:
```diff
@@ -175,6 +175,12 @@ int ENetPacketPeer::get_channels() const {
 }
 
+int ENetPacketPeer::get_packet_flags() const {
+	ERR_FAIL_NULL_V(peer, ERR_UNCONFIGURED);
+	ERR_FAIL_COND_V(packet_queue.is_empty(), ERR_UNAVAILABLE);
```
These are invalid returns, should be some flag, also don't think the `peer` part is necessary here

Faless commented on modules/enet/doc_classes/ENetPacketPeer.xml:
```diff
@@ -18,6 +18,12 @@
+		<method name="get_packet_flags" qualifiers="const">
+			<return type="int" />
+			<description>
+				Returns the ENet flags that were used to send the packet at the front of the queue. See [code]FLAG_*[/code] constants for available packet flags. Note that not all flags are replicated from the sending peer to the receiving peer.
```
```suggestion
				Returns the ENet flags of the next packet in the received queue. See [code]FLAG_*[/code] constants for available packet flags. Note that not all flags are replicated from the sending peer to the receiving peer.
```

Faless reviewed (approved):
LGTM, see my suggestion about the doc, nice work :rocket:

Chaosed0 committed:
Add method for getting packet flags from ENetPacketPeer

Chaosed0 commented:
Thanks for the heads up. I've rebased this commit onto the latest `master` branch and updated all my commits to be the correct author. Hopefully that is the right way to resolve things.

akien-mga commented:
It seems good, yes!

The final step for our PR workflow would be to squash the commits together. See [this page](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

I can also do it myself before merging tomorrow.

akien-mga commented:
Thanks!

akien-mga commented:
Thanks! And congrats for your first merged Godot contribution :tada:
